### PR TITLE
WIP: Nushell support

### PR DESCRIPTION
**Pull request by toadslop**
Hello @jdxcode . I have a draft of support for nushell. It's still a bit rough and doesn't completely handle the deactivate command and I also didn't confirm that the tests work, but I was hoping you could take a look and let me know if you have any suggestions or noticed anything that I missed other than the two things I mentioned above.

### CODE REVIEW

File: src/shell/nushell.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                    |before, after| _rtx_hook
+                  }}]
+              }}
+          }})
+            "#});
+
+        out
+    }
+
+    // TODO: properly handle deactivate
```
Comment (jdx) [inline src/shell/nushell.rs]: can you echo out this message for now?

```
echo "TODO: properly deactivate nushell"
```

we'll fix that later

Comment (toadslop) [reply src/shell/nushell.rs]: Sure.

Comment (toadslop) [reply src/shell/nushell.rs]: Do you mean that I should print it to the console when the command runs just to inform users that deactivate won't work properly?

Comment (jdx) [reply src/shell/nushell.rs]: Yes

Comment (toadslop) [reply src/shell/nushell.rs]: I added the line, but when I tested it for some reason RTX doesn't call Nushell's deactivate. It looks like it's calling zsh deactivate instead :/

Comment (toadslop) [reply src/shell/nushell.rs]: Ah, found the issue. Accidentally deleted my RTX_SHELL environment variable.

Comment (toadslop) [reply src/shell/nushell.rs]: Got deactivate working.

File: src/shell/mod.rs [hunk 2]
```diff
@@ -14,6 +15,7 @@ pub enum ShellType {
     Fish,
     Xonsh,
     Zsh,
+    Nu,
```
Comment (jdx) [inline src/shell/mod.rs]: Please alphabetize here and everywhere else

File: .tool-versions [hunk 3]
```diff
@@ -1,4 +0,0 @@
-shellcheck 0.9.0
```
Comment (jdx) [inline .tool-versions]: ?

File: README.md [hunk 4]
```diff
@@ -433,6 +433,12 @@ echo 'eval "$(rtx activate bash)"' >> ~/.bashrc
 echo 'rtx activate fish | source' >> ~/.config/fish/config.fish
 ```
 
+### Nushell
+
+```sh-session
+$ rtx activate nu | save $nu.config-path --append
```
Comment (jdx) [inline README.md]: I recently changed this in most other places, but remove the $ here

Comment (toadslop) [reply README.md]: Ok

**Comment by toadslop** [comment]
@jdxcode Added some changes to make the nushell activate script more robust. Please merge if there are no issues.

**Pull request merged**